contributors/design-proposals/storage/cloning.md

```diff
@@ -0,0 +1,158 @@
+
+## Background
+
+This document presents a proposal for defining best practices around security and implementation of cloning of persistent volumes in Kubernetes. 
+Cloning is distinctly different than storage snapshots in that it leverages the specific storage technology to ‘clone’ the physical volume 
+rather than copy the contents from one created PV to another. It also it not intended to capture a portion of data in time, 
+but always provide a full and complete copy.
+
+## Terminology
+* Clone - a duplicated volume created by the same storage technology as the original
+* Cloning - the process by which a storage technology can create a full copy of a volume
+
+## Goals
+* Provide consistent process to execute cloning for storage technologies both for out of tree plugins.
+
+## Non Goals
+* The process should not force storage vendors to implement cloning
+* Provide governance for security to clone assets between namespaces
+
+## Value add to Kubernetes  
+ By providing a consistent method for cloning, users can leverage the native capability of their storage to quickly create a 
+ copy of their persistent volumes without having to wait for a snapshot object to be created. This is especially desirable for large 
+ data volumes. Cloning also is a common use case in disaster recovery and for pre-seeding storage in a virtualized environment that many 
+ users accustomed to operating in those environments might find useful. As plug-ins move out of tree, it will be more difficult to 
+ enforce conformance to best practices. By creating an accepted framework for this process, we can ensure it’s done securely regardless 
+ of the storage provider.
+
+## Use Cases 
+Specific use cases around cloning are included in the use cases listed below as follows:
+* An admin wishes to create a copy of a volume to spawn another container to use the same data. 
+* An admin wishes to create a copy of a volume to pre-seed a process with information the container might need for bootstrapping.
+* A user wishes to clone a volume for purposes of disaster recovery
+
+## Design Overview
+The proposed design is based on the idea of leveraging well defined concepts for storage in Kubernetes. The cloning will be initiated 
+by creating an PVC to create a new volume using the dataSource API created with snapshots.
+The actual cloning process must be implemented by the provisioning storage and properly understand and react to the request to clone. 
+
+### API
+Leveraging the proposed API change for 'dataSource' the PVC will make a clone request by specifying the object it wishes to clone. In this example, we are cloning a pvc. For version 1 of cloning, we will limit cloning of pvcs to the current namespace only.
+
+** Pre-Request:
+```
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: pv-1
+  Namespace: myns
+spec:
+  capacity:
+    storage: 100Gi
+  volumeMode: Filesystem
+accessModes:
+    - ReadOnlyMany
```

Comment:
Do we need to think about restricting a clone for a RO source PV?

Reply:
@humblec I believe that came up in some discussions, but most devices that offer enhanced cloning features also handle things like dealing with consistency internally.  Even for those doing host assisted type cloning they would implicitly deal with this as well if they need to via mount options.  Also, just to be clear this is referencing the PVC and not the PV (just to make sure there's no confusion).

We don't want to restrict the cloning operation to RO I don't think, if we need something to control in-use PVCs we could perhaps use capabilities reporting (CSI Plugin/Spec) and deal with it there.  So something like "can clone in-use volumes" or something like that?